packages/core/src/app/shipping/PayPalFastlaneShippingAddress.tsx

```diff
@@ -22,20 +50,111 @@ export const PayPalFastlaneShippingAddress: FC<PayPalFastlaneShippingAddressProp
         deinitialize,
         isLoading,
         shippingAddress,
+        addresses,
+        isGuest,
+        handleFieldChange,
+        consignments
     } = props;
+    const paypalFastlaneShippingComponent = useRef<PayPalFastlaneAddressComponentRef>({});
 
-    return (
-        <PayPalFastlaneShippingAddressForm
-            address={shippingAddress}
-            countries={countries}
-            deinitialize={deinitialize}
-            formFields={formFields}
-            initialize={initialize}
-            isLoading={isLoading}
-            methodId={methodId}
-            onAddressSelect={onAddressSelect}
-            onFieldChange={onFieldChange}
-            onUnhandledError={onUnhandledError}
-        />
+    const paypalCommerceFastlaneOptions = {
+        paypalcommercefastlane: {
+            onPayPalFastlaneAddressChange: (
+                showPayPalFastlaneAddressSelector: PayPalFastlaneAddressComponentRef['showAddressSelector'],
+            ) => {
+                paypalFastlaneShippingComponent.current.showAddressSelector =
+                    showPayPalFastlaneAddressSelector;
+            },
+        },
+    };
+
+    const braintreeFastlaneOptions = {
+        braintreefastlane: {
+            onPayPalFastlaneAddressChange: (
+                showPayPalFastlaneAddressSelector: PayPalFastlaneAddressComponentRef['showAddressSelector'],
+            ) => {
+                paypalFastlaneShippingComponent.current.showAddressSelector =
+                    showPayPalFastlaneAddressSelector;
+            },
+        },
+    };
+
+    const initializationOptions: ShippingInitializeOptions = isPayPalCommerceFastlaneMethod(
+        methodId,
     )
+        ? paypalCommerceFastlaneOptions
+        : braintreeFastlaneOptions;
+
+    const initializeShippingStrategyOrThrow = async () => {
```

Comment:
Would be better to add some test for checking the strategy initialization and deinitialisation process.

Reply:
added